pkg/metrics/statsd/statsd.go

```diff
@@ -103,13 +107,67 @@ func (s *statsd) run() {
 	}
 }
 
-func (s *statsd) dial() (err error) {
-	path := "/var/etw/mdm_statsd.socket"
+func (s *statsd) parseSocketEnv(env string) (string, string, error) {
+	const (
+		malformed           = "malformed ENV variable ARO_STATSD_SOCKET. Expecting udp:<hostname>:<port> or unix:<path-to-socket> format. Got: %s"
+		unsupportedProtocol = "unsupported protocol in ENV variable ARO_STATSD_SOCKET. Expecting  'udp:' or 'unix:'. Got: %s in %s"
+		invalidUDPAddress   = "invalid UDP address in ENV variable ARO_STATSD_SOCKET %s. Error: %s "
+	)
+
+	// Verify protocol:connectionstring format
+	parameters := strings.SplitN(env, ":", 2)
+	if len(parameters) != 2 {
+		return "", "", fmt.Errorf(malformed, env)
+	}
+	protocol := parameters[0]
+	connectionstring := parameters[1]
+
+	//Verify supported protocol provided
+	if protocol != "udp" && protocol != "unix" {
+		return "", "", fmt.Errorf(unsupportedProtocol, protocol, env)
+	}
+
+	//UDP address check, no such (meaningful) thing for unix:
+	if protocol == "udp" {
+		_, err := net.ResolveUDPAddr(protocol, connectionstring)
+		if err != nil {
+			return "", "", fmt.Errorf(invalidUDPAddress, env, err)
+		}
+	}
```

Comment:
Is it necessary? Won't `net.Dial` return the same error? Any value in having a separate check here?

Reply:
for the error, maybe yes, but for the first part of what you selected, we have to check that it is not a weird protocol like tcp , ssh , telnet or whatever. This PR says that it enables things over UDP , and it should only allow UDP

Reply:
also, input sanitization is always a good practice.

Reply:
I think we can make good cases for f both approaches. I'm suggesting to leave it as is now.